### ISSUE TITLE
Clear Java DNS cache

### ISSUE THREAD

**Issue by Gitoffthelawn** [AdAway/AdAway#552]
In order to help remove the reboot requirement, clear the Java DNS cache after any change to the hosts file

**Comment by 0-kaladin** [comment]
I don't believe in android there is a good way to flush the dns cache system wide.  Potentially in a root shell this *could* do it: 
> ndc resolver flushdefaultif
but would have to be tested.

**Comment by Gitoffthelawn** [comment]
AdAway is excellent... the only weak spot is the need to reboot the device every time you make a change.

**Comment by Nick-White** [comment]
I agree.

**Comment by Gitoffthelawn** [comment]
Similar (but not identical) issue documented here: https://github.com/StevenBlack/hosts/pull/48
